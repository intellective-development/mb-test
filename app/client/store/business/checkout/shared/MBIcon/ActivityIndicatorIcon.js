import React from 'react';
import colors from '../MBElements/MBColors.css.json';

const ActivityIndicatorIcon = ({ color, ...props }) => (
  <svg
    version="1"
    viewBox="0 0 120 120"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
    {...props}>
    <defs>
      <path
        d="M60 11v16"
        id="l"
        stroke={color}
        strokeLinecap="round"
        strokeWidth="11">
        <animateTransform
          additive="sum"
          attributeName="transform"
          calcMode="discrete"
          dur="1s"
          keyTimes="0;.0833;.1666;.25;.3333;.4166;.5;.5833;.6666;.75;.8333;.9166;1"
          repeatCount="indefinite"
          type="rotate"
          /* eslint-disable-next-line max-len */
          values="0 60,60;30 60,60;60 60,60;90 60,60;120 60,60;150 60,60;180 60,60;210 60,60;240 60,60;270 60,60;300 60,60;330 60,60;360 60,60" />
      </path>
    </defs>
    <use
      opacity=".33"
      transform="rotate(30 60 60)"
      xlinkHref="#l" />
    <use
      opacity=".33"
      transform="rotate(60 60 60)"
      xlinkHref="#l" />
    <use
      opacity=".33"
      transform="rotate(90 60 60)"
      xlinkHref="#l" />
    <use
      opacity=".33"
      transform="rotate(120 60 60)"
      xlinkHref="#l" />
    <use
      opacity=".33"
      transform="rotate(150 60 60)"
      xlinkHref="#l" />
    <use
      opacity=".33"
      transform="rotate(180 60 60)"
      xlinkHref="#l" />
    <use
      opacity=".44"
      transform="rotate(210 60 60)"
      xlinkHref="#l" />
    <use
      opacity=".55"
      transform="rotate(240 60 60)"
      xlinkHref="#l" />
    <use
      opacity=".66"
      transform="rotate(270 60 60)"
      xlinkHref="#l" />
    <use
      opacity=".77"
      transform="rotate(300 60 60)"
      xlinkHref="#l" />
    <use
      opacity=".88"
      transform="rotate(330 60 60)"
      xlinkHref="#l" />
    <use
      xlinkHref="#l" />
  </svg>
);

ActivityIndicatorIcon.defaultProps = {
  color: colors.inlineIcon
};

export default ActivityIndicatorIcon;
