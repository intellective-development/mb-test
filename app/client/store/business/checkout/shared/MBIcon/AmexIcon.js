import React from 'react';

const AmexIcon = ({
  color,
  ...props
}) => (
  <svg
    version="1"
    viewBox="0 0 120 120"
    xmlns="http://www.w3.org/2000/svg"
    {...props}>
    <path
      d="M17 50L8 70h6l2-4h9l2 4h6l-9-20zm3 5h1l3 7h-7zm15 15V50h8l5 13 5-13h8v20h-5V56l-6 14h-4l-6-14v14zm32-20h17v4H72v4h12v4H72v4h12v4H67zm21 0h6l6 7 6-7h6l-9 10 9 10h-6l-6-7-6 7h-6l9-10z"
      fill={color} />
  </svg>
);

AmexIcon.defaultProps = {
  color: '#006fcf'
};

export default AmexIcon;
