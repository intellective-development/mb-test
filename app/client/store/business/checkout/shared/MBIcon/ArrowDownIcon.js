import React from 'react';
import colors from '../MBElements/MBColors.css.json';

const ArrowDownIcon = ({ color, ...props }) => (
  <svg
    version="1"
    viewBox="0 0 120 120"
    xmlns="http://www.w3.org/2000/svg"
    {...props}>
    <path
      d="M30 42l30 30 30-30"
      fill="none"
      stroke={color}
      strokeWidth="12" />
  </svg>
);

ArrowDownIcon.defaultProps = {
  color: colors.inlineIcon
};

export default ArrowDownIcon;
