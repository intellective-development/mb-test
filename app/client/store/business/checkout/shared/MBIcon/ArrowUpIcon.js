import React from 'react';
import colors from '../MBElements/MBColors.css.json';

const ArrowUpIcon = ({ color, ...props }) => (
  <svg
    version="1"
    viewBox="0 0 120 120"
    xmlns="http://www.w3.org/2000/svg"
    {...props}>
    <path
      d="M30 78l30-30 30 30"
      fill="none"
      stroke={color}
      strokeWidth="12" />
  </svg>
);

ArrowUpIcon.defaultProps = {
  color: colors.inlineIcon
};

export default ArrowUpIcon;
