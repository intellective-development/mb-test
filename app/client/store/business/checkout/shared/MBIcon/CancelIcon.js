import React from 'react';

const CancelIcon = ({ color, ...props }) => (
  <svg
    version="1"
    viewBox="0 0 120 120"
    xmlns="http://www.w3.org/2000/svg"
    {...props}>
    <circle
      cx="60"
      cy="60"
      fill={color}
      r="52" />
    <path
      d="M38 38l44 44m-44 0l44-44"
      stroke="#fff"
      strokeLinecap="round"
      strokeWidth="16" />
  </svg>
);

CancelIcon.defaultProps = {
  color: '#d9534f'
};

export default CancelIcon;
