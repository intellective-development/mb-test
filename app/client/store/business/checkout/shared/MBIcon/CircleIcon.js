import React from 'react';
import colors from '../MBElements/MBColors.css.json';

const CircleIcon = ({ color, ...props }) => (
  <svg
    version="1"
    viewBox="0 0 120 120"
    xmlns="http://www.w3.org/2000/svg"
    {...props}>
    <circle
      cx="60"
      cy="60"
      fill={color}
      r="50"
      strokeWidth="4" />
  </svg>
);

CircleIcon.defaultProps = {
  color: colors.inlineIcon
};

export default CircleIcon;
