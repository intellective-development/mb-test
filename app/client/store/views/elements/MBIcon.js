// @flow

import * as React from 'react';
import cn from 'classnames';

// This is intended to work as a unified location for our most common, simple ui icons (x's, arrows, etc.)
// As such, it is responsible for resizing 2x assets.

type IconName =
'close'
| 'back'
| 'check'
| 'minibar_logo'
| 'down_arrow_red'
| 'clock'
| 'pin'
| 'clear'
| 'mobile.minibar_logo'
| 'mobile.pencil';

type MBIconProps = {
  className?: string,
  name: IconName
};

const MBIcon = ({name, className, ...rest_props}: MBIconProps) => {
  const icon_fragment = formatIconFragment(name);
  const classes = cn(`el-mbicon--${name}`, className);

  return (
    <img
      alt={name}
      {...rest_props}
      className={classes}
      src={`/assets/components/elements/mb-icon/${icon_fragment}.png`}
      srcSet={`/assets/components/elements/mb-icon/${icon_fragment}@2x.png 2x, ` +
              `/assets/components/elements/mb-icon/${icon_fragment}@3x.png 3x`} />
  );
};

export default MBIcon;

// helpers
const formatIconFragment = (icon_name: string) => icon_name.replace('.', '/');
