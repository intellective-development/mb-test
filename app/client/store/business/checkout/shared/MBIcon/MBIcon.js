import { merge } from '@amory/style/umd/style';
import React from 'react';
import { renderToStaticMarkup } from 'react-dom/server';

import ActivityIndicatorIcon from './ActivityIndicatorIcon';
import AmexIcon from './AmexIcon';
import AppStoreIcon from './AppStoreIcon';
import ArrowDownIcon from './ArrowDownIcon';
import ArrowUpIcon from './ArrowUpIcon';
import CancelIcon from './CancelIcon';
import CircleIcon from './CircleIcon';
import ClockIcon from './ClockIcon';
import CompletedIcon from './CompletedIcon';
import DateIcon from './DateIcon';
import DiscoverIcon from './DiscoverIcon';
import DoneIcon from './DoneIcon';
import EmailIcon from './EmailIcon';
import { EnjoyIcon } from './EnjoyIcon';
import FacebookIcon from './FacebookIcon';
import { MapPinIcon } from './MapPinIcon';
import MastercardIcon from './MastercardIcon';
import MinibarLogo from './MinibarLogo';
import PhoneIcon from './PhoneIcon';
import PlayStoreIcon from './PlayStoreIcon';
import ProductIcon from './ProductIcon';
import { QuoteIcon } from './QuoteIcon';
import SecureIcon from './SecureIcon';
import { ShopIcon } from './ShopIcon';
import StorefrontIcon from './StorefrontIcon';
import TwitterIcon from './TwitterIcon';
import VisaIcon from './VisaIcon';
import VisibilityIcon from './VisibilityIcon';

const Icons = {
  amex: AmexIcon,
  appStore: AppStoreIcon,
  arrowDown: ArrowDownIcon,
  arrowUp: ArrowUpIcon,
  cancel: CancelIcon,
  circle: CircleIcon,
  clock: ClockIcon,
  completed: CompletedIcon,
  date: DateIcon,
  discover: DiscoverIcon,
  done: DoneIcon,
  email: EmailIcon,
  enjoy: EnjoyIcon,
  facebook: FacebookIcon,
  logo: MinibarLogo,
  mastercard: MastercardIcon,
  mapPin: MapPinIcon,
  phone: PhoneIcon,
  playStore: PlayStoreIcon,
  product: ProductIcon,
  quote: QuoteIcon,
  secure: SecureIcon,
  shop: ShopIcon,
  storefront: StorefrontIcon,
  twitter: TwitterIcon,
  visa: VisaIcon,
  visibility: VisibilityIcon,
  wait: ActivityIndicatorIcon
};

const onHover = ['clear', 'facebook', 'instagram', 'twitter'];

const MBIcon = ({
  name = 'circle',
  selector = '::before',
  style = {},
  ...props
}) => {
  const Icon = Icons[name] || Icons.circle;

  return merge({
    [selector]: {
      backgroundImage:
        `url("data:image/svg+xml,${encodeURIComponent(
          renderToStaticMarkup(<Icon {...props} />)
        )
          .replace(/%[\dA-F]{2}/gu, (match) => {
            switch (match){
              case '%20': return ' ';
              case '%3D': return '=';
              case '%3A': return ':';
              case '%2F': return '/';
              default: return match.toLowerCase();
            }
          })
        }"
      )`,
      backgroundRepeat: 'no-repeat',
      content: "' '",
      display: 'inline-block',
      minHeight: 16,
      minWidth: 16,
      ...style
    }
  },
  onHover.includes(name)
    ? {
      [`:hover${selector}`]: {
        backgroundImage: `url("data:image/svg+xml,${
          encodeURIComponent(
            renderToStaticMarkup(<Icon active {...props} />)
          )
            .replace(/%[\dA-F]{2}/gu, (match) => {
              switch (match){
                case '%20': return ' ';
                case '%3D': return '=';
                case '%3A': return ':';
                case '%2F': return '/';
                default: return match.toLowerCase();
              }
            })
        }")`
      }
    } : {});
};

export default MBIcon;
