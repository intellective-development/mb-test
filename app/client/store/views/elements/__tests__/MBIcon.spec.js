
import * as React from 'react';

// import TestProvider from 'client/store/__tests__/utils/TestProvider';
import MBIcon from '../MBIcon';

describe('MBIcon', () => {
  it('renders', () => {
    expect(shallow(<MBIcon name="back" />)).toMatchSnapshot();
  });

  it('renders a namespaced icon', () => {
    expect(shallow(<MBIcon name="mobile.minibar_logo" />)).toMatchSnapshot();
  });
});
