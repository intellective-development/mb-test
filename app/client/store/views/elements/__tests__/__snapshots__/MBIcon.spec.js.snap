// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MBIcon renders 1`] = `
<img
  alt="back"
  className="el-mbicon--back"
  src="/assets/components/elements/mb-icon/back.png"
  srcSet="/assets/components/elements/mb-icon/back@2x.png 2x, /assets/components/elements/mb-icon/back@3x.png 3x"
/>
`;

exports[`MBIcon renders a namespaced icon 1`] = `
<img
  alt="mobile.minibar_logo"
  className="el-mbicon--mobile.minibar_logo"
  src="/assets/components/elements/mb-icon/mobile/minibar_logo.png"
  srcSet="/assets/components/elements/mb-icon/mobile/minibar_logo@2x.png 2x, /assets/components/elements/mb-icon/mobile/minibar_logo@3x.png 3x"
/>
`;
